ConcurrentAuthzTest.testStaleCacheConcurrent intermittent failures

### Describe the bug

There seems to be intermittent failures when running ConcurrentAuthzTest.testStaleCacheConcurrent in GHA.

E.g. with this run: https://github.com/keycloak/keycloak/actions/runs/3249214665/jobs/5335974772 it failed with `legacy-jpa-federation+infinispan` with following stacktrace
```
2022-10-14T16:10:16.9692459Z [ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 4.067 s <<< FAILURE! - in org.keycloak.testsuite.model.authz.ConcurrentAuthzTest
2022-10-14T16:10:16.9697065Z [ERROR] org.keycloak.testsuite.model.authz.ConcurrentAuthzTest.testStaleCacheConcurrent  Time elapsed: 2.108 s  <<< ERROR!
2022-10-14T16:10:16.9698219Z org.keycloak.models.ModelDuplicateException: org.keycloak.models.ModelDuplicateException: javax.persistence.PersistenceException: org.hibernate.exception.ConstraintViolationException: could not execute statement
2022-10-14T16:10:16.9699327Z 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
2022-10-14T16:10:16.9700829Z 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
2022-10-14T16:10:16.9701613Z 	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
2022-10-14T16:10:16.9702231Z 	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
2022-10-14T16:10:16.9702748Z 	at java.base/java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:600)
2022-10-14T16:10:16.9703286Z 	at java.base/java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:678)
2022-10-14T16:10:16.9703779Z 	at java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:737)
2022-10-14T16:10:16.9704242Z 	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:159)
2022-10-14T16:10:16.9704723Z 	at java.base/java.util.stream.ForEachOps$ForEachOp$OfInt.evaluateParallel(ForEachOps.java:188)
2022-10-14T16:10:16.9705195Z 	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
2022-10-14T16:10:16.9705658Z 	at java.base/java.util.stream.IntPipeline.forEach(IntPipeline.java:439)
2022-10-14T16:10:16.9706083Z 	at java.base/java.util.stream.IntPipeline$Head.forEach(IntPipeline.java:596)
2022-10-14T16:10:16.9706677Z 	at org.keycloak.testsuite.model.authz.ConcurrentAuthzTest.testStaleCacheConcurrent(ConcurrentAuthzTest.java:153)
2022-10-14T16:10:16.9707311Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2022-10-14T16:10:16.9707866Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2022-10-14T16:10:16.9708477Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-10-14T16:10:16.9708987Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2022-10-14T16:10:16.9709420Z 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
2022-10-14T16:10:16.9709946Z 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
2022-10-14T16:10:16.9710492Z 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
2022-10-14T16:10:16.9711041Z 	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
2022-10-14T16:10:16.9711578Z 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
2022-10-14T16:10:16.9712082Z 	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
2022-10-14T16:10:16.9712628Z 	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
2022-10-14T16:10:16.9713027Z 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
2022-10-14T16:10:16.9713477Z 	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
2022-10-14T16:10:16.9713937Z 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
2022-10-14T16:10:16.9714415Z 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
2022-10-14T16:10:16.9714934Z 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
2022-10-14T16:10:16.9715384Z 	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
2022-10-14T16:10:16.9715811Z 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
2022-10-14T16:10:16.9716228Z 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
2022-10-14T16:10:16.9716656Z 	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
2022-10-14T16:10:16.9717064Z 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
2022-10-14T16:10:16.9717528Z 	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
2022-10-14T16:10:16.9717967Z 	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
2022-10-14T16:10:16.9718332Z 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
2022-10-14T16:10:16.9718728Z 	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
2022-10-14T16:10:16.9719260Z 	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:377)
2022-10-14T16:10:16.9719833Z 	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:284)
2022-10-14T16:10:16.9720418Z 	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:248)
2022-10-14T16:10:16.9720963Z 	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:167)
2022-10-14T16:10:16.9721503Z 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
2022-10-14T16:10:16.9722065Z 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
2022-10-14T16:10:16.9722559Z 	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
2022-10-14T16:10:16.9723043Z 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)
2022-10-14T16:10:16.9723748Z Caused by: org.keycloak.models.ModelDuplicateException: javax.persistence.PersistenceException: org.hibernate.exception.ConstraintViolationException: could not execute statement
2022-10-14T16:10:16.9724547Z 	at org.keycloak.connections.jpa.PersistenceExceptionConverter.convert(PersistenceExceptionConverter.java:80)
2022-10-14T16:10:16.9725211Z 	at org.keycloak.connections.jpa.JpaKeycloakTransaction.commit(JpaKeycloakTransaction.java:51)
2022-10-14T16:10:16.9725861Z 	at org.keycloak.services.DefaultKeycloakTransactionManager.commit(DefaultKeycloakTransactionManager.java:136)
2022-10-14T16:10:16.9726532Z 	at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransaction(KeycloakModelUtils.java:263)
2022-10-14T16:10:16.9727181Z 	at org.keycloak.testsuite.model.KeycloakModelTest.inComittedTransaction(KeycloakModelTest.java:543)
2022-10-14T16:10:16.9731115Z 	at org.keycloak.testsuite.model.KeycloakModelTest.inComittedTransaction(KeycloakModelTest.java:538)
2022-10-14T16:10:16.9731731Z 	at org.keycloak.testsuite.model.KeycloakModelTest.withRealm(KeycloakModelTest.java:566)
2022-10-14T16:10:16.9732366Z 	at org.keycloak.testsuite.model.authz.ConcurrentAuthzTest.lambda$testStaleCacheConcurrent$8(ConcurrentAuthzTest.java:180)
2022-10-14T16:10:16.9732937Z 	at java.base/java.util.stream.ForEachOps$ForEachOp$OfInt.accept(ForEachOps.java:204)
2022-10-14T16:10:16.9733407Z 	at java.base/java.util.stream.Streams$RangeIntSpliterator.forEachRemaining(Streams.java:104)
2022-10-14T16:10:16.9733875Z 	at java.base/java.util.Spliterator$OfInt.forEachRemaining(Spliterator.java:699)
2022-10-14T16:10:16.9734339Z 	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
2022-10-14T16:10:16.9734917Z 	at java.base/java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:290)
2022-10-14T16:10:16.9735391Z 	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:746)
2022-10-14T16:10:16.9735864Z 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
2022-10-14T16:10:16.9736346Z 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
2022-10-14T16:10:16.9736810Z 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
2022-10-14T16:10:16.9737272Z 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
2022-10-14T16:10:16.9737777Z 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
2022-10-14T16:10:16.9738396Z Caused by: javax.persistence.PersistenceException: org.hibernate.exception.ConstraintViolationException: could not execute statement
2022-10-14T16:10:16.9739030Z 	at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:154)
2022-10-14T16:10:16.9739597Z 	at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:181)
2022-10-14T16:10:16.9740220Z 	at org.hibernate.internal.ExceptionConverterImpl.convertCommitException(ExceptionConverterImpl.java:65)
2022-10-14T16:10:16.9740862Z 	at org.hibernate.engine.transaction.internal.TransactionImpl.commit(TransactionImpl.java:104)
2022-10-14T16:10:16.9741535Z 	at org.keycloak.connections.jpa.JpaKeycloakTransaction.commit(JpaKeycloakTransaction.java:49)
2022-10-14T16:10:16.9741935Z 	... 17 more
2022-10-14T16:10:16.9742306Z Caused by: org.hibernate.exception.ConstraintViolationException: could not execute statement
2022-10-14T16:10:16.9742909Z 	at org.hibernate.exception.internal.SQLExceptionTypeDelegate.convert(SQLExceptionTypeDelegate.java:59)
2022-10-14T16:10:16.9743621Z 	at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:37)
2022-10-14T16:10:16.9744262Z 	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:113)
2022-10-14T16:10:16.9744843Z 	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:99)
2022-10-14T16:10:16.9745460Z 	at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:200)
2022-10-14T16:10:16.9746111Z 	at org.hibernate.engine.jdbc.batch.internal.NonBatchingBatch.addToBatch(NonBatchingBatch.java:46)
2022-10-14T16:10:16.9746753Z 	at org.hibernate.persister.entity.AbstractEntityPersister.delete(AbstractEntityPersister.java:3698)
2022-10-14T16:10:16.9747440Z 	at org.hibernate.persister.entity.AbstractEntityPersister.delete(AbstractEntityPersister.java:3958)
2022-10-14T16:10:16.9748040Z 	at org.hibernate.action.internal.EntityDeleteAction.execute(EntityDeleteAction.java:123)
2022-10-14T16:10:16.9748561Z 	at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:604)
2022-10-14T16:10:16.9749058Z 	at org.hibernate.engine.spi.ActionQueue.lambda$executeActions$1(ActionQueue.java:478)
2022-10-14T16:10:16.9749510Z 	at java.base/java.util.LinkedHashMap.forEach(LinkedHashMap.java:684)
2022-10-14T16:10:16.9749968Z 	at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:475)
2022-10-14T16:10:16.9750618Z 	at org.hibernate.event.internal.AbstractFlushingEventListener.performExecutions(AbstractFlushingEventListener.java:344)
2022-10-14T16:10:16.9751343Z 	at org.hibernate.event.internal.DefaultFlushEventListener.onFlush(DefaultFlushEventListener.java:40)
2022-10-14T16:10:16.9752085Z 	at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:107)
2022-10-14T16:10:16.9752701Z 	at org.hibernate.internal.SessionImpl.doFlush(SessionImpl.java:1407)
2022-10-14T16:10:16.9753166Z 	at org.hibernate.internal.SessionImpl.managedFlush(SessionImpl.java:489)
2022-10-14T16:10:16.9753722Z 	at org.hibernate.internal.SessionImpl.flushBeforeTransactionCompletion(SessionImpl.java:3290)
2022-10-14T16:10:16.9754410Z 	at org.hibernate.internal.SessionImpl.beforeTransactionCompletion(SessionImpl.java:2425)
2022-10-14T16:10:16.9755103Z 	at org.hibernate.engine.jdbc.internal.JdbcCoordinatorImpl.beforeTransactionCompletion(JdbcCoordinatorImpl.java:449)
2022-10-14T16:10:16.9756243Z 	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl.beforeCompletionCallback(JdbcResourceLocalTransactionCoordinatorImpl.java:183)
2022-10-14T16:10:16.9757577Z 	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl.access$300(JdbcResourceLocalTransactionCoordinatorImpl.java:40)
2022-10-14T16:10:16.9758823Z 	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl$TransactionDriverControlImpl.commit(JdbcResourceLocalTransactionCoordinatorImpl.java:281)
2022-10-14T16:10:16.9759742Z 	at org.hibernate.engine.transaction.internal.TransactionImpl.commit(TransactionImpl.java:101)
2022-10-14T16:10:16.9760139Z 	... 18 more
2022-10-14T16:10:16.9761282Z Caused by: org.h2.jdbc.JdbcSQLIntegrityConstraintViolationException: Referential integrity constraint violation: "FK_FRSR5S213XCX4WNKOG82SSRFY: PUBLIC.ASSOCIATED_POLICY FOREIGN KEY(ASSOCIATED_POLICY_ID) REFERENCES PUBLIC.RESOURCE_SERVER_POLICY(ID) ('c447ba40-3555-4c6a-8a4a-a3221a908f86')"; SQL statement:
2022-10-14T16:10:16.9762086Z delete from RESOURCE_SERVER_POLICY where ID=? [23503-214]
2022-10-14T16:10:16.9762550Z 	at org.h2.message.DbException.getJdbcSQLException(DbException.java:508)
2022-10-14T16:10:16.9763017Z 	at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
2022-10-14T16:10:16.9763424Z 	at org.h2.message.DbException.get(DbException.java:223)
2022-10-14T16:10:16.9763769Z 	at org.h2.message.DbException.get(DbException.java:199)
2022-10-14T16:10:16.9764207Z 	at org.h2.constraint.ConstraintReferential.checkRow(ConstraintReferential.java:365)
2022-10-14T16:10:16.9764754Z 	at org.h2.constraint.ConstraintReferential.checkRowRefTable(ConstraintReferential.java:382)
2022-10-14T16:10:16.9765300Z 	at org.h2.constraint.ConstraintReferential.checkRow(ConstraintReferential.java:256)
2022-10-14T16:10:16.9765739Z 	at org.h2.table.Table.fireConstraints(Table.java:1172)
2022-10-14T16:10:16.9766089Z 	at org.h2.table.Table.fireAfterRow(Table.java:1190)
2022-10-14T16:10:16.9766415Z 	at org.h2.command.dml.Delete.update(Delete.java:92)
2022-10-14T16:10:16.9766831Z 	at org.h2.command.dml.DataChangeStatement.update(DataChangeStatement.java:74)
2022-10-14T16:10:16.9767287Z 	at org.h2.command.CommandContainer.update(CommandContainer.java:169)
2022-10-14T16:10:16.9767792Z 	at org.h2.command.Command.executeUpdate(Command.java:252)
2022-10-14T16:10:16.9768278Z 	at org.h2.jdbc.JdbcPreparedStatement.executeUpdateInternal(JdbcPreparedStatement.java:209)
2022-10-14T16:10:16.9768831Z 	at org.h2.jdbc.JdbcPreparedStatement.executeUpdate(JdbcPreparedStatement.java:169)
2022-10-14T16:10:16.9769417Z 	at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:197)
2022-10-14T16:10:16.9769850Z 	... 38 more
```

### Version

19

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Anything else?

_No response_

--- a/testsuite/model/src/test/java/org/keycloak/testsuite/model/authz/ConcurrentAuthzTest.java
+++ b/testsuite/model/src/test/java/org/keycloak/testsuite/model/authz/ConcurrentAuthzTest.java
@@ -17,6 +17,7 @@
 
 package org.keycloak.testsuite.model.authz;
 
+import org.junit.Ignore;
 import org.junit.Test;
 import org.keycloak.authorization.AuthorizationProvider;
 import org.keycloak.authorization.model.Policy;
@@ -133,6 +134,7 @@ public class ConcurrentAuthzTest extends KeycloakModelTest {
     }
 
     @Test
+    @Ignore // This is ignored due to intermittent failure, see https://github.com/keycloak/keycloak/issues/14917
     public void testStaleCacheConcurrent() {
         String permissionId = withRealm(realmId, (session, realm) -> {
             AuthorizationProvider authorization = session.getProvider(AuthorizationProvider.class);

